react/features/invite/components/InviteButton.web.js

```diff
@@ -0,0 +1,231 @@
+     * Handles selection of the invite options.
+     *
+     * @param { Object } option - The invite option that has been selected from
+     * the dropdown menu.
+     * @private
+     * @returns {void}
+     */
+    _onInviteOptionSelected(option) {
+        this.state.inviteOptions[0].items.forEach(item => {
+            if (item.content === option.item.content) {
+                item.action();
+            }
+        });
+    }
+
+    /**
+     * Updates the invite items list depending on the availability of the
+     * features.
+     *
+     * @param {Object} props - The read-only properties of the component.
+     * @private
+     * @returns {void}
+     */
+    _updateInviteItems(props) {
+        const { t } = this.props;
+
+        const inviteItems = [];
+
+        inviteItems.splice(
+            getInviteOptionPosition(SHARE_LINK_OPTION),
+            0,
+            {
+                content: t('toolbar.invite'),
+                action: () => this.props.openDialog(InviteDialog)
+            }
+        );
+
+        if (props._isDialOutAvailable) {
+            inviteItems.splice(
+                getInviteOptionPosition(DIAL_OUT_OPTION),
+                0,
+                {
+                    content: t('dialOut.dialOut'),
+                    action: () => this.props.openDialog(DialOutDialog)
+                }
+            );
+        }
+
+        if (props._isAddToCallAvailable) {
+            inviteItems.splice(
+                getInviteOptionPosition(ADD_TO_CALL_OPTION),
+                0,
+                {
+                    content: interfaceConfig.ADD_PEOPLE_APP_NAME,
+                    action: () => this.props.openDialog(AddPeopleDialog)
+                }
+            );
+        }
+
+        this.state = {
+            /**
+             * The list of invite options.
+             */
+            inviteOptions: [
+                {
+                    items: inviteItems
+                }
+            ]
+        };
+    }
+}
+
+/**
+ * Maps (parts of) the Redux state to the associated {@code InviteButton}'s
+ * props.
+ *
+ * @param {Object} state - The Redux state.
+ * @private
+ * @returns {{
+ *     _isAddToCallAvailable: boolean,
+ *     _isDialOutAvailable: boolean
+ * }}
+ */
+function _mapStateToProps(state) {
+    const { enableUserRolesBasedOnToken } = state['features/base/config'];
+
+    const { conference } = state['features/base/conference'];
+
+    const { isGuest } = state['features/jwt'];
+
+    return {
+        _isAddToCallAvailable: !isGuest
+            && isInviteOptionEnabled(ADD_TO_CALL_OPTION),
+        _isDialOutAvailable:
+            getLocalParticipant(state).role === PARTICIPANT_ROLE.MODERATOR
+            && conference && conference.isSIPCallingSupported()
+            && isInviteOptionEnabled(DIAL_OUT_OPTION)
+            && (!enableUserRolesBasedOnToken || !isGuest)
+    };
+}
+
+export default translate(connect(
+    _mapStateToProps, { openDialog })(InviteButton));
```

Comment:
I've seen you use this pattern before where you map an action to a prop. What are the advantages of this approach?

Reply:
I like it because I don't need to call dispatch and it shortens my code when calling the action.